internal/applicationsnapshot/input_test.go

```diff
@@ -34,6 +40,72 @@ import (
 	"github.com/enterprise-contract/ec-cli/internal/utils"
 )
 
+type MockRemoteClient struct {
```

Comment:
We use github.com/stretchr/testify/mock in some places to assist in mocking, example https://github.com/enterprise-contract/ec-cli/blob/ca3cf7bd6a864d57cc2edf8e33e1ffd296f7c4f3/internal/fetchers/oci/fake/client.go#L80

See if makes sense here.